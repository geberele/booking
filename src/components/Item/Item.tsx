import { IResults } from '../../api/search'
import React, { useState } from 'react'
import classNames from 'classnames'

interface IItemProps {
  result: IResults
  onClick: (value: string) => void
}

export const Item = ({ result, onClick }: IItemProps) => {
  const { name, placeType, city, country, iata, region } = result
  const [classOnFocus, setClassOnFocus] = useState<string>('')
  const locationName = [name, iata].filter(el => el).join(' ')
  const locationCity = [city, region, country].filter(el => el).join(', ')

  const handleEventOnActive = () => setClassOnFocus('item--focus')
  const handleEventOnInactive = () => setClassOnFocus('')
  const handleOnClick = () => onClick(locationName)
  const handleOnKeyDown = (e: React.KeyboardEvent) => {
    if (e.key === 'Enter') {
      onClick(locationName)
    }
  }

  return (
    <li
      className={classNames('item', classOnFocus)}
      tabIndex={0}
      onMouseOver={handleEventOnActive}
      onMouseLeave={handleEventOnInactive}
      onFocus={handleEventOnActive}
      onBlur={handleEventOnInactive}
      aria-label={name}
      onClick={handleOnClick}
      onKeyDown={handleOnKeyDown}
    >
      {placeType &&
        <>
          <div className="item__place">
            <span className="item__pill">{placeType}</span>
          </div>
          <div className="item__location">
            <span className="item__location-name">{locationName}</span>
            <span className="item__location-city">{locationCity}</span>
          </div>
        </>
       }
      {!placeType &&
        <div className="item__place">
          {name}
        </div>
      }
    </li>
  )
}
