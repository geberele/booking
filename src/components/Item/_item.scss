.item {
  background: $color-white;
  border-top: 1px solid $color-grey-2;
  color: $color-grey-3;
  cursor: pointer;
  display: inline-block;
  float: left;
  padding: 9px 0;
  width: 100%;

  &:first-child {
    border: none;
  }

  &--focus {
    background: $color-blu-2;
  }

  &:focus {
    outline: none;
  }

  &__place {
    float: left;
    padding: 8px 16px;
  }

  &__pill {
    background-color: $color-blu;
    border-radius: 4px;
    color: $color-white;
    display: inline-block;
    font-family: $font-main;
    font-size: 12px;
    line-height: 1.25rem;
    min-width: 60px;
    padding: 1px 4px;
    text-align: center;
  }

  &__location {
    display: flex;
    flex-direction: column;
    float: left;
    padding: 0;

    &-name {
      font-size: 14px;
      line-height: 1.25rem;
    }

    &-city {
      font-size: 12px;
      line-height: 1.25rem;
    }
  }
}
