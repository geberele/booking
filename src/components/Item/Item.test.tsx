import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { Item, IItemProps } from './Item'

describe('Item', () => {
  it('renders with all the props', () => {
    const props: IItemProps = {
      result: {
        country: 'Italy',
        city: 'Milan',
        region: 'Lombardy',
        name: 'Milan Malpensa Airport',
        iata: 'MXP',
        placeType: 'Airport',
      },
      onClick: () => {
        return
      },
    }
    const { asFragment } = render(<Item {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders with "No results found"', () => {
    const props: IItemProps = {
      result: {
        country: '',
        city: '',
        region: '',
        name: 'No results found',
        iata: '',
        placeType: '',
      },
      onClick: () => {
        return
      },
    }
    const { asFragment } = render(<Item {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
