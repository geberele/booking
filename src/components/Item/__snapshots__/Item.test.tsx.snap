// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders with "No results found" 1`] = `
<DocumentFragment>
  <li
    aria-label="No results found"
    class="item"
    tabindex="0"
  >
    <div
      class="item__place"
    >
      No results found
    </div>
  </li>
</DocumentFragment>
`;

exports[`Item renders with all the props 1`] = `
<DocumentFragment>
  <li
    aria-label="Milan Malpensa Airport"
    class="item"
    tabindex="0"
  >
    <div
      class="item__place"
    >
      <span
        class="item__pill"
      >
        Airport
      </span>
    </div>
    <div
      class="item__location"
    >
      <span
        class="item__location-name"
      >
        Milan Malpensa Airport MXP
      </span>
      <span
        class="item__location-city"
      >
        Milan, Lombardy, Italy
      </span>
    </div>
  </li>
</DocumentFragment>
`;
